src/app/edit-user-name/edit-user-name.component.ts

```diff
@@ -0,0 +1,78 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {OverlayService} from "../../shared/services/overlay.service";
+import {AbstractControl, FormControl, FormGroup, Validators} from "@angular/forms";
+import {ErrorMessageService} from "../../shared/services/error-message.service";
+import {UserService} from "../../shared/services/user.service";
+import {AuthService} from "../../shared/services/auth.service";
+import {User} from "../../shared/types/User";
+import {Router} from "@angular/router";
+
+@Component({
+  selector: 'app-edit-user-name',
+  templateUrl: './edit-user-name.component.html',
+  styleUrls: ['./edit-user-name.component.scss']
+})
+export class EditUserNameComponent implements OnInit{
+  submitted: boolean = false;
```

Comment:
Should be using @input and @output from angular core to handle communication between this component and the parent `profile.component.html`. Would look something like this:

```
export class EditUserNameComponent implements OnInit {
  @Input() isOpen: boolean = false;
  @Output() closeModal: new EventEmitter<void>();

  submitted: boolean = false;
  .... etc ....
```